tests/fixtures/ecma-version/13/top-level-await/top-await.result.js

```diff
@@ -0,0 +1,183 @@
+export default {
+    "type": "Program",
+    "loc": {
+        "start": {
+            "line": 1,
+            "column": 0
+        },
+        "end": {
+            "line": 1,
+            "column": 12
+        }
+    },
+    "range": [
+        0,
+        12
+    ],
+    "body": [
+        {
+            "type": "ExpressionStatement",
+            "loc": {
+                "start": {
+                    "line": 1,
+                    "column": 0
+                },
+                "end": {
+                    "line": 1,
+                    "column": 12
+                }
+            },
+            "range": [
+                0,
+                12
+            ],
+            "expression": {
+                "type": "AwaitExpression",
+                "loc": {
+                    "start": {
+                        "line": 1,
+                        "column": 0
+                    },
+                    "end": {
+                        "line": 1,
+                        "column": 11
+                    }
+                },
+                "range": [
+                    0,
+                    11
+                ],
+                "argument": {
+                    "type": "CallExpression",
+                    "loc": {
+                        "start": {
+                            "line": 1,
+                            "column": 6
+                        },
+                        "end": {
+                            "line": 1,
+                            "column": 11
+                        }
+                    },
+                    "range": [
+                        6,
+                        11
+                    ],
+                    "callee": {
+                        "type": "Identifier",
+                        "loc": {
+                            "start": {
+                                "line": 1,
+                                "column": 6
+                            },
+                            "end": {
+                                "line": 1,
+                                "column": 9
+                            }
+                        },
+                        "range": [
+                            6,
+                            9
+                        ],
+                        "name": "foo"
+                    },
+                    "arguments": [],
+                    "optional": false
+                }
+            }
+        }
+    ],
+    "sourceType": "script",
```

Comment:
Shouldn't `await foo();` be a parsing error with `"sourceType": "script"`, top-level await is allowed only in modules?

Reply:
This also works, but I think it should throw a parsing error instead:

```js
import * as espree from "./espree.js";

const ast = espree.parse("class foo { bar = await baz; }", { ecmaVersion: 2022, sourceType: "module" });

console.log(ast.body[0].body.body[0].value.type); // AwaitExpression
```

It looks like `ecmaVersion: 2022` in Acorn just enables the `allowAwaitOutsideFunction` option, which doesn't check the context. https://github.com/acornjs/acorn/commit/7405bd13c45718d8da02c2f2079b46ef78d2410b

Reply:
Yeah, this should be a parsing error, you’re right. I can update the test sourceType but otherwise we need to open an issue in Acorn.

Reply:
Reported in https://github.com/acornjs/acorn/issues/1048

Reply:
Looks like the fix was committed today. I'll merge in the next Acorn release when it drops.

Reply:
Updated the test.